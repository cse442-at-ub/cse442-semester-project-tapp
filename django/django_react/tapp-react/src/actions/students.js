import axios from 'axios';

import { GET_EVENTS, ADD_EVENT, GET_ERRORS } from './types';

//GET STUDENTS
export const getEvents = (param) => dispatch => {
  axios.get('http://localhost:3000/api/class?classNum='+param)
  .then(res => {
    dispatch({
      type: GET_EVENTS,
      payload: res.data
    });
  })
};


//ADD USER
export const addEvent = events => dispatch => {
  axios	
  .post('/api/class', events)
  .then(res => {
  dispatch({
    type: ADD_EVENT,
    payload: res.data
  });
  })
  .catch(err => {
  dispatch({
    type: GET_ERRORS, 
    payload: err
    });
  });
};
