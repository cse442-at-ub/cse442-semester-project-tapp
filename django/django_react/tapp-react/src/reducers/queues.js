import {GET_QUEUES, GET_INSTR, ADD_QUEUE, DELETE_QUEUE} from "../actions/types.js"

const initialState = {
  queues: [],
}

export default function(state=initialState, action) {
  switch(action.type) {
    case GET_QUEUES:
      return {
      ...state,
      queues: action.payload
      };
    case DELETE_QUEUE:
      return {
      ...state,
      queues: state.queues.filter((even) => even.id !== action.payload)
      };
    case ADD_QUEUE:
      return {
      ...state,
      queues: [...state.queues, action.payload]};
    default: return state
  }
}
