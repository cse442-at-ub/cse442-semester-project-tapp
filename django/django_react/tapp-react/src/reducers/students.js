import {GET_EVENTS, GET_INSTR, ADD_EVENT, DELETE_EVENT} from "../actions/types.js"

const initialState = {
  events: [],
  instructors: []
}

export default function(state=initialState, action) {
  switch(action.type) {
    case GET_EVENTS:
      return {
      ...state,
      events: action.payload
      };
    case GET_INSTR:
      return {
      ...state,
      instructors: action.payload
      };
    case DELETE_EVENT:
      return {
      ...state,
      events: state.events.filter((even) => even.id !== action.payload)
      };
    case ADD_EVENT:
      return {
      ...state,
      events: [...state.events, action.payload]};
    default: return state
  }
}
