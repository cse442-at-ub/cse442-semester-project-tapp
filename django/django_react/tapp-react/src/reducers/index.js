import {combineReducers} from 'redux';
import students from './students'
import errors from './errors'
import auth from './auth'

export default combineReducers({
  students,
  auth,
  errors
});
