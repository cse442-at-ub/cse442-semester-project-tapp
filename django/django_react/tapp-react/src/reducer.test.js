import reducer from '../src/reducers/students'
import {initialState} from '../src/reducers/students'

describe( 'user get reducer', () => {
  it('should return the initialState', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })
  
  it('should handle GET_STUDENTS', () => {
    expect(
      reducer(initialState,
      {
        type: 'GET_STUDENTS'})).toMatchSnapshot()})
})
