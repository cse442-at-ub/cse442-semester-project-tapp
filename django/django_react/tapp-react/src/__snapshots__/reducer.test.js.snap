// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user get reducer should handle GET_STUDENTS 1`] = `
Object {
  "students": undefined,
}
`;

exports[`user get reducer should return the initialState 1`] = `
Object {
  "students": Array [],
}
`;
